packages/bitcore-node/src/modules/ethereum/api/internalTxTransform.ts

```diff
@@ -0,0 +1,33 @@
+import { Transform } from 'stream';
+import Web3 from 'web3';
+import { MongoBound } from '../../../models/base';
+import { WalletAddressStorage } from '../../../models/walletAddress';
+import { IEthTransaction } from '../types';
+
+export class InternalTxRelatedFilterTransform extends Transform {
+  constructor(private web3: Web3, private walletId) {
+    super({ objectMode: true });
+  }
+
+  async _transform(tx: MongoBound<IEthTransaction>, _, done) {
+    if (tx.internal && tx.internal.length > 0) {
+      const walletAddresses = await WalletAddressStorage.collection
```

Comment:
should pull getting the wallet addresses into a method, and have it cache them, so you don't have to hit the db per tx


```
getWalletAddresses() {
 if(!this.walletAddresses.length){
    this.walletAddresses = await find()
  }
  return this.walletAddresses
}
```

Reply:
Done. Thanks !